### PR TITLE
fix authorizer setting and use getpermissionprefix

### PR DESCRIPTION
**Pull request by noodlemoodle**
https://nanopay.atlassian.net/browse/NP-10353

### CODE REVIEW

File: src/foam/dao/MaterializedDAO.js [hunk 1]
```diff
@@ -153,7 +152,7 @@ foam.CLASS({
       synchronized: true,
       javaCode: `
         if ( ! getInitialized() ) {
-          Thread t = new Thread(this);
+Thread t = new Thread(this);
```
Comment (kgrgreer) [inline src/foam/dao/MaterializedDAO.js]: Why this change?

File: src/foam/dao/MaterializedDAO.js [hunk 2]
```diff
@@ -204,7 +203,7 @@ foam.CLASS({
       javaCode: `
         try {
           getQueue().put(new Object[] { PUT, value });
-        } catch (InterruptedException e) {
+          } catch (InterruptedException e) {
```
Comment (kgrgreer) [inline src/foam/dao/MaterializedDAO.js]: Why this change?

File: src/foam/dao/MaterializedDAO.js [hunk 3]
```diff
@@ -217,7 +216,7 @@ foam.CLASS({
       javaCode: `
         try {
           getQueue().put(new Object[] { REMOVE, value });
-        } catch (InterruptedException e) {
+} catch (InterruptedException e) {
```
Comment (kgrgreer) [inline src/foam/dao/MaterializedDAO.js]: Why this change?

File: src/foam/dao/MaterializedDAO.js [hunk 4]
```diff
@@ -230,11 +229,11 @@ foam.CLASS({
         getQueue().clear();
         try {
           getQueue().put(new Object[] { REMOVE_ALL });
-        } catch (InterruptedException e) {
+} catch (InterruptedException e) {
```
Comment (kgrgreer) [inline src/foam/dao/MaterializedDAO.js]: Why this change?

File: src/foam/dao/MaterializedDAO.js [hunk 5]
```diff
@@ -230,11 +229,11 @@ foam.CLASS({
...
         }
         return this;
       `
-    },
+},
```
Comment (kgrgreer) [inline src/foam/dao/MaterializedDAO.js]: Why this change?

Comment (noodlemoodle) [inline src/foam/dao/MaterializedDAO.js]:
```suggestion
    },
```

File: src/foam/dao/MaterializedDAO.js [hunk 6]
```diff
@@ -230,7 +229,7 @@ foam.CLASS({
         getQueue().clear();
         try {
           getQueue().put(new Object[] { REMOVE_ALL });
-        } catch (InterruptedException e) {
+} catch (InterruptedException e) {
```
Comment (noodlemoodle) [inline src/foam/dao/MaterializedDAO.js]:
```suggestion
        } catch (InterruptedException e) {
```

File: src/foam/dao/MaterializedDAO.js (hunk 3 above)
Comment (noodlemoodle) [inline src/foam/dao/MaterializedDAO.js]:
```suggestion
        } catch (InterruptedException e) {
```

File: src/foam/dao/MaterializedDAO.js (hunk 2 above)
Comment (noodlemoodle) [inline src/foam/dao/MaterializedDAO.js]:
```suggestion
        } catch (InterruptedException e) {
```

File: src/foam/dao/MaterializedDAO.js (hunk 1 above)
Comment (noodlemoodle) [inline src/foam/dao/MaterializedDAO.js]:
```suggestion
        Thread t = new Thread(this);
```

File: src/foam/dao/MaterializedDAO.js [hunk 7]
```diff
@@ -153,7 +152,7 @@ foam.CLASS({
       synchronized: true,
       javaCode: `
         if ( ! getInitialized() ) {
-          Thread t = new Thread(this);
+        Thread t = new Thread(this);
```
Comment (noodlemoodle) [inline src/foam/dao/MaterializedDAO.js]:
```suggestion
          Thread t = new Thread(this);
```

File: src/foam/dao/MaterializedDAO.js [hunk 8]
```diff
@@ -43,7 +43,7 @@ foam.CLASS({
   `,
 
   properties: [
-    {
+{
```
Comment (noodlemoodle) [inline src/foam/dao/MaterializedDAO.js]:
```suggestion
    {
```

File: src/foam/dao/MaterializedDAO.js [hunk 9]
```diff
@@ -97,11 +97,10 @@ foam.CLASS({
       type: 'foam.nanos.auth.Authorizer',
       name: 'authorizer',
       javaFactory: `
-      String sourceClass = getSourceDAO().getOf().getObjClass().getSimpleName().toLowerCase();
-      if ( foam.nanos.auth.Authorizable.class.isAssignableFrom(getOf().getObjClass()) ) {
-        return new foam.nanos.auth.AuthorizableAuthorizer(sourceClass);
+      if ( foam.nanos.auth.Authorizable.class.isAssignableFrom(getSourceDAO().getOf().getObjClass()) ) {
```
Comment (lchanmann) [inline src/foam/dao/MaterializedDAO.js:100]: @noodlemoodle `getSourceDAO().getOf().getObjClass()` may not be the same as `getOf().getObjClass()`, the `of` of the mat dao could be different from the `of` of the source dao.

The AuthorizableAuthorizer should only apply when the mat dao `of` implements Authorizable.
